### DISCUSSION TITLE
Route guard

### DISCUSSION THREAD

**Discussion by ghost** [antfu/vitesse#410] [Q&A] [answered]
Any way to put a route guard ?
Old questions of the discussion doesn't answered:(

**Comment by NamesMT** [comment, marked as answer]
After taking a peek at `vite-plugin-pages`, I think we could use [custom metadata](https://github.com/hannoeru/vite-plugin-pages#sfc-custom-block-for-route-data) and [`extendRoute`](https://github.com/hannoeru/vite-plugin-pages#extendroute) to implement guards.

Edit:
Actually, `extendRoute` is not even needed, you could use the custom metadata along with a beforeEach that handles everything.
It is, however, needed if you want to add per-route guards (`beforeEnter`).

**Reply by MarcoSantana** [reply]
Please can you post an example?
Thanks!!!

**Reply by mergeMarc** [reply]
Here is a quick example i created. Please note that i'm new to Vue so there is probably better/securer ways to check and store if the user is authenticated. In my example i used a Pina store to save authentication information (`useAuthStore`).

I created file src/modules/guard.ts:
```ts
import { type UserModule } from '~/types'

export const install: UserModule = ({ isClient, router }) => {
    router.beforeEach((to, from) => {
        const auth = useAuthStore()
        //             Page requires login     Prevent redirect loops       User authentication check
        if(isClient && to.meta.requiresAuth && to.path !== '/app/login'  && !auth.isAuthenticated()) {
            console.log('Authentication required for page \'' + to.name?.toString() + '\'! Redirecting to login.')
            // Redirect user to login page
            return { path: '/app/login' }
        }
    })
}
```

In the pages (under src/pages) that require the user to be logged in, i added the meta tag 'requiresAuth: true' to the route block.
Here is an example of my file src/pages/app/index.vue:
```vue
<script setup>
</script>

<template>
    [App - Home]
</template>

<route lang="yaml">
meta:
  layout: home
  requiresAuth: true
</route>
```
